pkg/client/fake/client.go

```diff
@@ -329,6 +338,31 @@ func (t versionedTracker) Update(gvr schema.GroupVersionResource, obj runtime.Ob
 	return t.ObjectTracker.Update(gvr, obj, ns)
 }
 
+func isCustomResource(gvk schema.GroupVersionKind) bool {
+	return !scheme.Scheme.Recognizes(gvk)
+}
+
+func didNonMetadataFieldChange(old runtime.Object, new runtime.Object) bool {
+	oldUnstructured, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(old)
+	newUnstructured, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(new)
+
+	oldCopyContent := copyNonMetadata(oldUnstructured)
+	newCopyContent := copyNonMetadata(newUnstructured)
+
+	return !apiequality.Semantic.DeepEqual(oldCopyContent, newCopyContent)
+}
```

Comment:
```suggestion
func didNonMetadataFieldChange(old runtime.Object, new runtime.Object) bool {
 oldCopy, newCopy := old.DeepCopy(), new.DeepCop()
 oldCopy.Metadata = metav1.ObjectMeta{}
 newCopy.Metadata = metav1.ObjectMeta{}
 return !reflect.DeepEqual(oldCopy, newCopy)
}
```

Reply:
This approach doesn't work, as you can't access the `Metadata` from the `runtime.Object` interface. Reverting the refactor.